### PR TITLE
`azurerm_snapshot` adding `network_access_policy` and `public_network_access`

### PR DESCRIPTION
**Pull request by rizkybiz**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description
This change introduces `network_access_policy` and `public_network_access` parameters.

## PR Checklist

- [] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [ ] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [ ] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [ ] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

* `azurerm_snapshot` - support for the `network_access_policy` property
* `azurerm_snapshot` - support for the `public_network_access` property

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @rizkybiz thanks for opening this. This looks good to me, but could we add update steps for these tests too? Thanks!

**Comment by rizkybiz** [comment]
@catriona-m Made the changes above. Let me know if there's any additional changes needed.

File: internal/services/compute/snapshot_resource_test.go [hunk 1]
```diff
@@ -245,11 +273,13 @@ resource "azurerm_managed_disk" "test" {
...
-  resource_group_name = azurerm_resource_group.test.name
-  create_option       = "Copy"
-  source_uri          = azurerm_managed_disk.test.id
+  name                  = "acctestss_%d"
+  location              = azurerm_resource_group.test.location
+  resource_group_name   = azurerm_resource_group.test.name
+  create_option         = "Copy"
+  source_uri            = azurerm_managed_disk.test.id
+  network_access_policy = "AllowAll"
+  public_access         = "Enabled"
```
Comment (catriona-m) [inline internal/services/compute/snapshot_resource_test.go:282]:
```suggestion
  public_network_access         = "Enabled"
```

File: website/docs/r/snapshot.html.markdown [hunk 2]
```diff
@@ -65,6 +65,10 @@ The following arguments are supported:
 
 * `incremental_enabled` - (Optional) Specifies if the Snapshot is incremental. Changing this forces a new resource to be created.
 
+* `network_access_policy` - (Optional) Policy for accessing the disk via network. Possible values are `AllowAll`, `AllowPrivate`, or `DenyAll`. Default: `AllowAll`.
+
+* `public_network_access` - (Optional) Policy for controlling export on the disk. Possible values are `Enabled` or `Disabled`. Default: `Enabled`.
+
```
Comment (catriona-m) [inline website/docs/r/snapshot.html.markdown:71]: For consistency with other properties, could we specify the defaults as below instead?

```suggestion
* `network_access_policy` - (Optional) Policy for accessing the disk via network. Possible values are `AllowAll`, `AllowPrivate`, or `DenyAll`. Defaults to `AllowAll`.

* `public_network_access` - (Optional) Policy for controlling export on the disk. Possible values are `Enabled` or `Disabled`. Defaults to `Enabled`.

```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @rizkybiz thanks for your work on this. I have left two minor comments inline, but otherwise this should be good to go once those are fixed. Thanks!

**Pull request merged**